src/solver/utils/ortools_utils.cpp

```diff
@@ -9,6 +9,63 @@
 using namespace operations_research;
 
 const char* const XPRESS_PARAMS = "THREADS 1";
+enum class MATRIX_ELEMENT_TYPE
+{
+    VARIABLE,
+    CONSTRAINT
+};
+
+char FromMatrixElementTypeToChar(MATRIX_ELEMENT_TYPE type)
+{
+    if (type == MATRIX_ELEMENT_TYPE::VARIABLE)
+    {
+        return 'x';
+    }
+    return 'c';
+}
+
+void CheckName(std::string& var_name, unsigned idxVar, MATRIX_ELEMENT_TYPE type)
+{
+    if (var_name.empty())
+    {
+        var_name = FromMatrixElementTypeToChar(type) + std::to_string(idxVar);
+    }
+}
+
+static void UpdateCoefficients(MPSolver* solver,
+                               MPObjective* const objective,
+                               const double* bMin,
+                               const double* bMax,
+                               const double* costs,
+                               unsigned idxVar,
+                               std::string& varName,
+                               MATRIX_ELEMENT_TYPE type)
+{
+    double min_l = 0.0;
+    if (bMin != NULL)
+    {
+        min_l = bMin[idxVar];
+    }
+    double max_l = bMax[idxVar];
+    CheckName(varName, idxVar, type);
+    const MPVariable* var = solver->MakeNumVar(min_l, max_l, varName);
+    objective->SetCoefficient(var, costs[idxVar]);
+}
+
+static void transferVariables(MPSolver* solver,
+                              const double* bMin,
+                              const double* bMax,
+                              const double* costs,
+                              int nbVar)
+{
+    MPObjective* const objective = solver->MutableObjective();
+    for (int idxVar = 0; idxVar < nbVar; ++idxVar)
+    {
+        std::string varName;
+        UpdateCoefficients(
+          solver, objective, bMin, bMax, costs, idxVar, varName, MATRIX_ELEMENT_TYPE::VARIABLE);
```

Comment:
You may wrap this behavior into a namer, taking an `std::vector` as ctor argument.

```cpp
class VariableNamer {
VariableNamer(const std::vector<string>& names);
std::string getName(unsigned index);
};

static void transferVariables(MPSolver* solver,
                              const double* bMin,
                              const double* bMax,
                              const double* costs,
                              int nbVar
                              VariableNamer& namer);

```

Then you only need one version for `transferVariables` and `transferRows`, not 2.